src/plugin/oauth/index.js

```diff
@@ -2075,61 +2075,62 @@ class BaseOauthPlugin extends BasePlugin {
      */
     if (
       pluginStrategy == PLUGIN_STRATEGY_OIDC &&
-      plugin.config.assertions.aud &&
-      idToken.aud != plugin.config.client.client_id
+      plugin.config.assertions.aud
     ) {
-      return false;
+      if (idToken.aud != plugin.config.client.client_id) {
+        plugin.server.logger.verbose("aud does not match client_id");
+        return false;
+      }
+    } else {
+      plugin.server.logger.verbose("Assertions: SKIPPED aud")
     }
 
     /**
      * access token is expired and refresh tokens are disabled
      */
-    if (
-      plugin.config.assertions.exp &&
-      tokenset_is_expired(tokenSet) &&
-      !(
-        plugin.config.features.refresh_access_token &&
-        tokenset_can_refresh(tokenSet)
-      )
-    ) {
-      plugin.server.logger.verbose(
-        "tokenSet is expired and refresh tokens disabled"
-      );
-      return false;
-    }
+    if (plugin.config.assertions.exp) {
+      const is_expired = tokenset_is_expired(tokenSet)
 
-    /**
-     * both access and refresh tokens are expired and refresh is enabled
-     */
-    if (
-      plugin.config.assertions.exp &&
-      tokenset_is_expired(tokenSet) &&
-      plugin.config.features.refresh_access_token &&
-      !tokenset_can_refresh(tokenSet)
-    ) {
-      plugin.server.logger.verbose(
-        "tokenSet expired and refresh no longer available"
-      );
-      return false;
+      if (is_expired) {
+        plugin.server.logger.verbose("tokenSet is expired");
+
+        if (!plugin.config.features.refresh_access_token) {
+          plugin.server.logger.verbose("refresh tokens disabled");
+          return false
+        }
+
+        if (!tokenset_can_refresh(tokenSet)) {
+          plugin.server.logger.verbose("refresh no longer available");
+          return false
+        }
+        
+        plugin.server.logger.verbose("refresh token available");
+      }
+      plugin.server.logger.verbose("Assertions: PASSED exp");
+    } else {
+      plugin.server.logger.verbose("Assertions: SKIPPED exp")
     }
 
-    if (
-      pluginStrategy == PLUGIN_STRATEGY_OIDC &&
```

Comment:
This check needs to remain.

Reply:
Can you elaborate on this? We use OAuth2 and need it to validate `nbf` and `iss`.